src/sandbox/wiktor/LineChart.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useRef } from 'react';
+import {
+  LineChart,
+  Line,
+  XAxis,
+  YAxis,
+  CartesianGrid,
+  Tooltip,
+  Legend,
+  ResponsiveContainer,
+} from 'recharts';
+
+interface ILineChartProps {
+  data: { datetime: string; value: number }[];
+}
+
+const LineChartComponent: React.FC<ILineChartProps> = ({ data }) => {
+  const chartRef = useRef<HTMLDivElement>(null);
+
+  const minValue = Math.min(...data.map((d) => d.value));
+  const maxValue = Math.max(...data.map((d) => d.value));
+  const margin = 0.02;
+  const yDomain = [
+    Math.round(minValue * (1.0 - margin)),
+    Math.round(maxValue * (1.0 + margin)),
+  ];
+
+  return (
+    <div ref={chartRef} style={{ height: '300px' }}>
+      <ResponsiveContainer width="100%" height="100%" debounce={1}>
+        <LineChart data={data} margin={{ bottom: 60, left: 20, right: 20 }}>
+          <XAxis dataKey="datetime" angle={-90} textAnchor="end" interval={0} />
+          <YAxis domain={yDomain} />
+          <CartesianGrid stroke="#eee" strokeDasharray="5 5" />
+          <Tooltip />
+          <Legend verticalAlign="top" align="right" />
```

Comment:
Aaa no i jeszcze tutaj ta legenda przydałaby się ładniejsza, bo "value" może nie mówić za dużo userowi.
Zerknij tutaj: https://recharts.org/en-US/api/Legend#content

Widziałbym to tak: 
High price (color primary.light)
Average price (color primary.main)
Low price (color primary.dark)